cli/internal/transformerpipeline/pipeline.go

```diff
@@ -81,16 +86,24 @@ func (lp *TransformerPipeline) OnOutput(fn func([]byte) error) error {
 	return nil
 }
 
-func (lp *TransformerPipeline) Close() error {
+func (lp *TransformerPipeline) Close() {
+	// Closing the pipeline happens on both source as well as destination close.
+	// Not handling this will result in a close of closed channel panic.
+	if lp.clientWrappers[0].isClosed {
+		return
+	}
+
 	// Close the first transformer. The rest will follow gracefully, otherwise records will be lost.
-	return lp.clientWrappers[0].client.CloseSend()
+	lp.clientWrappers[0].client.CloseSend()
```

Comment:
```suggestion
	_ = lp.clientWrappers[0].client.CloseSend()
```

You might need this to tell the linter you're ignoring the error from `CloseSend()`

Reply:
But you can ignore this if the linter passes